salt/states/svn.py

```diff
@@ -95,13 +95,27 @@ def latest(name,
                      'The path "{0}" exists and is not '
                      'a directory.'.format(target)
                      )
+    try:
+        current_info = __salt__['svn.info'](cwd, target, user=user, username=username, password=password, fmt='dict')
+        svn_cmd = 'svn.update'
+    except exceptions.CommandExecutionError:
```

Comment:
Hey Mike,

It was not problematic in itself (svn_cmd was declared before the exception and it would do the right thing if exception was caught), but while reviewing my code, I discovered another bug when the target directory existed but was not a svn working copy.

I added an exception statement within the test block now, it should make the code clearer.

If target directory exists and is not a working copy, it fails:

```
          ID: svn_repo
    Function: svn.latest
        Name: file:///home/svn
      Result: False
     Comment: /var/www/svn exists but is not a svn working copy.
     Started: 10:43:08.289164
    Duration: 17.355 ms
     Changes:
```

If the directory doesn't exist and is set to be checked out, it now returns the revision it is going to check out:

```
          ID: svn_repo
    Function: svn.latest
        Name: file:///home/svn
      Result: None
     Comment: /var/www/svn doesn't exist and is set to be checked out at revision 4.
     Started: 10:35:23.961889
    Duration: 0.498 ms
     Changes:
```

If directory is set to be updated it gives:

```
          ID: svn_repo
    Function: svn.latest
        Name: file:///home/svn
      Result: None
     Comment: Index: /var/www/svn/README.md
              ===================================================================
              --- /var/www/svn/README.md	(revision 3)
              +++ /var/www/svn/README.md	(revision 4)
              @@ -1,3 +1,3 @@
               initial import
               -------------
              -rev 3
              +rev 4
     Started: 10:45:42.755722
    Duration: 34.862 ms
     Changes:
```